src/plugins/intel_gpu/src/graph/graph_optimizer/pre_dynamic_shape_opts.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "pass_manager.h"
+#include "gather_inst.h"
+#include "permute_inst.h"
+#include "kv_cache_inst.h"
+#include "gemm_inst.h"
+#include "program_helpers.h"
+
+using namespace cldnn;
+
+void pre_dynamic_shape_opts::run(program& p) {
```

Comment:
Suggestion: How about "mark_runtime_skippable_nodes" for better understanding of  the purpose?

Reply:
updated as suggested